toolkit/tools/imagegen/configuration/networkconfig_test.go

```diff
@@ -134,17 +134,30 @@ func TestShouldFailParsingInvalidDevice_Network(t *testing.T) {
 }
 
 func TestShouldPassCreatingNetworkFile_Network(t *testing.T) {
-	const networkFile = "/etc/systemd/network/10-static-eth1.network"
+	const (
+		networkFileDir  = "/etc/systemd/network"
+		testNetworkFile = "/etc/systemd/network/10-static-eth1.network"
+	)
+
+	// Some systems may not have systemd-networkd service configured, and thus
+	// /etc/systemd/network may not exist. For this test case, manually create this directory
+	// if it does not exist
+	exists, err := file.DirExists(networkFileDir)
+	if err != nil || !exists {
+		err = os.Mkdir(networkFileDir, os.ModePerm)
+		assert.NoError(t, err)
+	}
```

Comment:
I think we can skip directly to trying to create it with [MkDirAll](https://pkg.go.dev/os#MkdirAll), since it doesn't error out if the directory exists:
```suggestion
	err := os.MkDirAll(networkFileDir)
	assert.NoError(t, err)
```

Reply:
Scratch that. We should actually clean up after ourselves, if the directory didn't exist before the tests:
```suggestion
	exists, err := file.DirExists(networkFileDir)
	assert.NoError(t, err)
	if !exists {
		err = os.Mkdir(networkFileDir, os.ModePerm)
		assert.NoError(t, err)
		t.Cleanup(func() {
			err = os.RemoveAll(networkFileDir)
			assert.NoError(t, err)
		})
	}
```

Please double-check if this compiles, if you decide to use it - it's just an illustration of the idea.;)